### PR TITLE
Improve performance little

### PR DESCRIPTION
**Pull request by ksh8281**
* divide Value::toBoolean into 2 functions for better function call inlinling
* try to avoid call RopeString::flattenRopeString if possible
  flatten RopeString for read one char is waste I think

Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

File: src/runtime/ValueInlines.h [hunk 1]
```diff
@@ -807,34 +807,7 @@ inline bool Value::toBoolean(ExecutionState& ec) const // $7.1.2 ToBoolean
...
-    }
-
-#if defined(ESCARGOT_ENABLE_TEST)
-    if (UNLIKELY(checkIfObjectWithIsHTMLDDA())) {
-        return false;
-    }
-#endif
-    // Symbol, Objects..
-    return true;
+    return toBooleanSlowCase(ec);
```
Comment (bbrto21) [inline src/runtime/ValueInlines.h:810]: Q. How does this lead to performance improvement?

**Review by bbrto21 — APPROVED** [review]
LGTM

File: src/runtime/ValueInlines.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/ValueInlines.h:810]: the shorter Value::toBoolean can be inline-able to compilers.
the longer version of Value::toBoolean is not inlined by compilers.

Comment (bbrto21) [reply src/runtime/ValueInlines.h:810]: Thanks you for answer!
I remembered the godbolt you presented at the seminar last time in best reviewer course work.
so I tested it using this, I can see [only short function is inlined by gcc](https://godbolt.org/z/hY7bqPbM5)

File: src/runtime/StaticStrings.h [hunk 2]
```diff
@@ -825,6 +825,14 @@ class StaticStrings {
         free(numbers);
     }
 
+    Escargot::String* charCodeToString(char32_t ch)
+    {
+        if (ch < ESCARGOT_ASCII_TABLE_MAX) {
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/StaticStrings.h:830]: Good Point!

File: src/runtime/StringObject.cpp [hunk 3]
```diff
@@ -25,11 +25,7 @@ namespace Escargot {
 
 static inline Value charToStringValue(ExecutionState& state, char16_t ch)
 {
-    if (ch < ESCARGOT_ASCII_TABLE_MAX) {
-        return state.context()->staticStrings().asciiTable[ch].string();
-    } else {
-        return String::fromCharCode(ch);
-    }
+    return state.context()->staticStrings().charCodeToString(ch);
 }
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/StringObject.cpp:29]: I see

**Review by clover2123 — APPROVED** [review]
LGTM 👍

**Pull request merged**